### DISCUSSION TITLE
Error conntecting

### DISCUSSION THREAD

**Discussion by ehlen000** [KelvinTegelaar/CIPP#1281] [Q&A]
Hi all,
I think Cipp looks really great, something we all have been waiting for for many years.  so I deployed Cipp in my Azure Visual Studio Subcription and I can access the web frontend. Then I used the manual SAM setup, followed all guidelines but when i run the permissions check it says: 

invalid_grant:AADSTS9002313

I do have some trouble understanding wat the "refresh token" and "exchange refresh token" lines should contain since I cannot find any additional info on those. I tried "empty" and I tried using the token info which I found in the configuration of the SAM app. 
I followed all troubleshooting steps and cleared the token cache but no luck. Can this be related to the Visual Studio subscription? or am I doing something else wrong.

Does anyone have tips or tricks on how to proceed on this? 

greetz....

**Comment by KelvinTegelaar** [comment]
Check the documentation at https://cipp.app and follow it to the letter.

**Reply by ehlen000** [reply]
thanks for the quick response. I think I did, but I will start again, maybe i did miss a letter :-)
